backend/src/message/message.gateway.ts

```diff
@@ -62,4 +78,21 @@ export class MessageGateway {
     console.log('data lastViewTime : ', data.lastViewTime);
     // console.log(typeof data.lastViewTime);
   }
+
+  /**
+   * private
+   */
+  private async checkExistFriendship(friendId: number, senderId: number, receiverId: number): Promise<void> {
+    const friendship = await this.friendshipRepository.findOneBy({ id: friendId });
+    if (friendship === null) {
+      throw new Error('존재하지 않는 친구 관계입니다.');
+    }
+    if (
+      (friendship.senderId === senderId && friendship.receiverId === receiverId && friendship.accept === true) ||
+      (friendship.senderId === receiverId && friendship.receiverId === senderId && friendship.accept === true)
+    ) {
+      return;
+    }
+    throw new Error('유효하지 않은 친구 관계입니다.');
+  }
```

Comment:
```suggestion
    const friendship = await this.friendshipRepository.findOneBy([
      { senderId: senderId, receiverId: receiverId }, // sender -> receiver
      { senderId: receiverId, receiverId: senderId }, // receiver -> sender
    ]);
      if (friendship === null || friendship.id !== friendId || friendship.accept !== true) {
      throw new Error('유효하지 않은 친구 관계입니다.');
      }
```

어떠신가용??

Reply:
이거 고려했었는데 `pk`만 사용한 단일조건으로 `select`하고 얻어온 값으로 직접 나머지 `value`를 비교하면 유효성 검사가 더 빨리 동작하지 않을까 라는 생각으로 이렇게 작성했습니다! 근데 실제로 어떻게 비교해야할 지몰라서 정말 얼마나 차이나는지는 모르겠어요..ㅠㅜ

Reply:
```suggestion
    const friendship = await this.friendshipRespository.findOneBy([
  { id: friendId, senderId: senderId, receiverId: receiverId, accept: true },
  { id: friendId, senderId: receiverId, receiverId: senderId, accept: true },
]);
    if (friendship === null) {
      throw new Error('존재하지 않는 친구 관계입니다.');
    }
    if (
      (friendship.senderId === senderId && friendship.receiverId === receiverId && friendship.accept === true) ||
      (friendship.senderId === receiverId && friendship.receiverId === senderId && friendship.accept === true)
    ) {
      return;
    }
    throw new Error('유효하지 않은 친구 관계입니다.');
  }
```

로 하는 건 더 오래 걸리나요?? 이렇게하면 완전 간단해질 거 같긴한데..ㅎㅎ

Reply:
는 오래 걸릴 것 같군요..ㅎ

Reply:
속도를 고려해서 id로 찾는 방법으로 진행하신다면 이대로 갈 수 밖에 없겠는데요??